staging/src/k8s.io/component-base/codec/codec.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package codec
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// NewLenientSchemeAndCodecs constructs a CodecFactory with strict decoding
+// disabled, that has only the Schemes registered into it which are passed
+// and added via AddToScheme functions. This can be used to skip strict decoding
+// a specific version only.
+func NewLenientSchemeAndCodecs(addToSchemeFns ...func(s *runtime.Scheme) error) (*runtime.Scheme, *serializer.CodecFactory, error) {
```

Comment:
this introduced function seems not referenced by others at all; in other words, dead code. 
consider either embracing changes from loadConfig (as in the original PR), or remove this code change.

Reply:
What is the community PR you are refering to? During cherry-pick, I prefer to pick all changes together, not by selection as we never know what is required later.

Reply:
itook second look; the code is actually called by cmd scheduler app module's loadConfig function; not dead code. The change is good. sorry for the msreading.